src/components/FilterByNumber.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useContext, useState } from 'react';
+import PlanetsContext from '../context/Planets';
+
+function FilterByNumber() {
+  const INITIAL_FILTER = {
+    column: 'population',
+    comparison: 'maior que',
+    value: '0',
+  };
+
+  const INITIAL_COLUMN_OPTIONS = [
+    'population',
+    'orbital_period',
+    'diameter',
+    'rotation_period',
+    'surface_water',
+  ];
+
+  const { handleFilterValues,
+    filterByNumericValues,
+    deleteNumericFilter,
+  } = useContext(PlanetsContext);
+
+  const [numericFilter, setNumericFilter] = useState(INITIAL_FILTER);
+  const [columnOptions, setColumnOptions] = useState(INITIAL_COLUMN_OPTIONS);
+
+  const handleChange = ({ target }) => {
+    setNumericFilter({
+      ...numericFilter,
+      [target.name]: target.value,
+    });
+  };
+
+  const submitFilterValues = () => {
+    handleFilterValues(numericFilter);
+    const newOptions = [...columnOptions];
+    newOptions.splice(columnOptions.indexOf(numericFilter.column), 1);
+    setColumnOptions(newOptions);
```

Comment:
```suggestion
    setColumnOptions(columnOptions.filter(option => option !== numericFilter.column));
```

use HOF's, so u dont need to use shallow copy on the array

Reply:
much simpler too, thanks for the tip